app/src/main/java/com/io/moviO/domain/GetMoviesUseCase.kt

```diff
@@ -1,11 +1,18 @@
 package com.io.moviO.domain
 
+import com.io.moviO.data.DataResult
 import com.io.moviO.data.Movie
 import com.io.moviO.data.MoviesRepository
 
-class GetMoviesUseCase {
+class GetMoviesUseCase : UseCase<MoviesRepository, DataResult<List<Movie>>> {
```

Comment:
The repository should be a dependency on the class. You use it to get the movies, but it is not a required data to get them. A required data would be an id, year, etc.